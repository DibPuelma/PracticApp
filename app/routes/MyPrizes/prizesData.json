{
  "prizes": [
    {
      "prize":"$100.000",
      "store":"Americanino",
      "logo":"http://www.borde.cl/wp-content/uploads/2005/11/Americanino-2.png",
      "code":"187297"
    },
    {
      "prize":"$35.000",
      "store":"Umbrale",
      "logo":"http://static03.ofertia.cl/comercios/Umbrale/profile-850.v11.png",
      "code":"346512"
    },
    {
      "prize":"$1.000.000",
      "store":"Samsung",
      "logo":"http://www.brycomm.com/wp-content/uploads/2011/08/Samsung-Logo.png",
      "code":"534562"
    },
    {
      "prize":"$10.000",
      "store":"Apple",
      "logo":"http://pelfind.me/images/187941/apple-logo-9th.jpg",
      "code":"754332"
    },
    {
      "prize":"$55.000",
      "store":"Native Instruments",
      "logo":"http://emblemsbf.com/img/7808.jpg",
      "code":"467853"
    }
  ]
}
