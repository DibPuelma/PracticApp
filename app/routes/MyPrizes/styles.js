import {
  StyleSheet,
  Dimensions
} from 'react-native'

module.exports = StyleSheet.create({
  container:{
    marginTop: 72,
    marginLeft: 10,
    marginBottom: 10,
    marginRight: 10
  },
  listElement: {
    flex: 2,
    flexDirection: 'row',
    width: Dimensions.get('window').width,
    padding: 2,
    borderBottomWidth: 1,
    borderColor: '#969696'
  },
  image: {
    flex: 1,
    height: 80,
  },
  prize: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#E6E6E6'
  },
  prizeText: {
    fontSize: 25
  }
})
