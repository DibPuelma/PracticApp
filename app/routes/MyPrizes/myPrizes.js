import React, { Component } from 'react';
import {
  View,
  ListView,
  Text,
  Image,
  ActivityIndicator,
  TouchableHighlight
} from 'react-native';

import prizesData from './prizesData';
import styles from './styles';
import backButtonHandler from '../../lib/backButtonHandler';
import settings from '../../config/settings';

var MyPrizesStatus = {
  WAITING: 'waiting',
  READY  : 'ready'
};

export default class MyPrizes extends Component{
  constructor(props) {
    super(props);
    const ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 !== r2});
    this.state = {
      status: MyPrizesStatus.WAITING,
      dataSource: ds.cloneWithRows([])
    };

    this._backToPrevious = this._backToPrevious.bind(this);
    this.singletonBackButtonHandler = backButtonHandler.getInstance();
  }

  componentDidMount(){
    this._userPrizesRequest();
  }

  componentWillMount(){
    this.singletonBackButtonHandler.addBackEvent(this._backToPrevious);
  }

  componentWillUnmount() {
    this.singletonBackButtonHandler.removeBackEvent(this._backToPrevious);
  }

  _backToPrevious() {
    this.props.navigator.replace({id:'QRReader'});
    return true; // This is important to prevent multiple calls
  }

  render() {
    if (this.state.status === MyPrizesStatus.WAITING) {
      return (
      <View style={styles.loadingContainer}>
        <ActivityIndicator animating={true} style={[styles.centering, {height: 50}]} size="large" color="#3FA9F5"/>
      </View>
      )
    }
    return (
      <View style={styles.container}>
        <ListView
          dataSource={this.state.dataSource}
          renderRow={(rowData) => (
              <TouchableHighlight onPress={() => this._goToDetails(rowData)} >
                <View style={styles.listElement}>
                  <View style={{ flex: 1 }}>
                    <Image source={{uri: rowData.Contest.Company.logo}} style={styles.image} />
                  </View>
                  <View style={styles.info}>
                    <Text numberOfLines={3} style={styles.companyName} textAlign='left'>
                      {rowData.Contest.Company.name}
                    </Text>
                    <Text numberOfLines={3} style={styles.prize} textAlign='left'>
                      {rowData.name}
                    </Text>
                  </View>
                </View>
              </TouchableHighlight>
            )}
          enableEmptySections={true}
        />
      </View>
    );
  }

  _userPrizesRequest() {
    var user_id = this.props.user.id;
    var url = settings.USER_PRIZES_REQUEST.replace(":id", user_id);
    var promise = fetch(url, {
      method: 'GET',
      headers: {
        'Accept': 'application/json',
        'Content-Type': 'application/json',
      }, 
    })
    .then((response) => response.json());
    
    var myPrizes = this;

    promise.then(function(result) {
      console.log(result);      
      myPrizes.setState({ status: myPrizes.READY });

      const ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 !== r2}); 
      myPrizes.setState({ dataSource: ds.cloneWithRows(result) });
    }, function(err) { // error
      console.log(err);    
    });
  }

  _goToDetails(rowData){
    this.props.navigator.push({id:'PrizeDetails', prizeData: rowData})
  }
}
