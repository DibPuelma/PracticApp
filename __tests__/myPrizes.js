import 'react-native';
import React from 'react';
import MyPrizes from '../app/routes/MyPrizes/myPrizes'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <MyPrizes />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
