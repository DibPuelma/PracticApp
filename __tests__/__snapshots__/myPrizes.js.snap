exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "marginBottom": 10,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 72,
    }
  }>
  <ScrollView
    dataSource={
      ListViewDataSource {
        "items": 5,
      }
    }
    initialListSize={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={1000}
    onKeyboardDidHide={undefined}
    onKeyboardDidShow={undefined}
    onKeyboardWillHide={undefined}
    onKeyboardWillShow={undefined}
    onLayout={[Function]}
    onScroll={[Function]}
    pageSize={1}
    removeClippedSubviews={true}
    renderRow={[Function]}
    scrollEventThrottle={50}
    scrollRenderAheadDistance={1000}
    stickyHeaderIndices={Array []}>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}>
      <View
        style={
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#969696",
            "flex": 2,
            "flexDirection": "row",
            "padding": 2,
            "width": 750,
          }
        }>
        <Image
          source={
            Object {
              "uri": "http://www.borde.cl/wp-content/uploads/2005/11/Americanino-2.png",
            }
          }
          style={
            Object {
              "flex": 1,
              "height": 80,
            }
          } />
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#E6E6E6",
              "flex": 1,
              "justifyContent": "center",
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={3}
            style={
              Object {
                "fontSize": 25,
              }
            }>
            $100.000
          </Text>
        </View>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}>
      <View
        style={
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#969696",
            "flex": 2,
            "flexDirection": "row",
            "padding": 2,
            "width": 750,
          }
        }>
        <Image
          source={
            Object {
              "uri": "http://static03.ofertia.cl/comercios/Umbrale/profile-850.v11.png",
            }
          }
          style={
            Object {
              "flex": 1,
              "height": 80,
            }
          } />
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#E6E6E6",
              "flex": 1,
              "justifyContent": "center",
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={3}
            style={
              Object {
                "fontSize": 25,
              }
            }>
            $35.000
          </Text>
        </View>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}>
      <View
        style={
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#969696",
            "flex": 2,
            "flexDirection": "row",
            "padding": 2,
            "width": 750,
          }
        }>
        <Image
          source={
            Object {
              "uri": "http://www.brycomm.com/wp-content/uploads/2011/08/Samsung-Logo.png",
            }
          }
          style={
            Object {
              "flex": 1,
              "height": 80,
            }
          } />
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#E6E6E6",
              "flex": 1,
              "justifyContent": "center",
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={3}
            style={
              Object {
                "fontSize": 25,
              }
            }>
            $1.000.000
          </Text>
        </View>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}>
      <View
        style={
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#969696",
            "flex": 2,
            "flexDirection": "row",
            "padding": 2,
            "width": 750,
          }
        }>
        <Image
          source={
            Object {
              "uri": "http://pelfind.me/images/187941/apple-logo-9th.jpg",
            }
          }
          style={
            Object {
              "flex": 1,
              "height": 80,
            }
          } />
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#E6E6E6",
              "flex": 1,
              "justifyContent": "center",
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={3}
            style={
              Object {
                "fontSize": 25,
              }
            }>
            $10.000
          </Text>
        </View>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}>
      <View
        style={
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#969696",
            "flex": 2,
            "flexDirection": "row",
            "padding": 2,
            "width": 750,
          }
        }>
        <Image
          source={
            Object {
              "uri": "http://emblemsbf.com/img/7808.jpg",
            }
          }
          style={
            Object {
              "flex": 1,
              "height": 80,
            }
          } />
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#E6E6E6",
              "flex": 1,
              "justifyContent": "center",
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={3}
            style={
              Object {
                "fontSize": 25,
              }
            }>
            $55.000
          </Text>
        </View>
      </View>
    </View>
  </ScrollView>
</View>
`;
